src/main/java/com/google/sps/algorithms/MatchQuery.java

```diff
@@ -35,16 +36,28 @@
     }
 
     for(User potentialMentor : allUsers) {
-        Collection<String> mentorSpecialties = potentialMentor.getSpecialties();
+      Collection<String> mentorSpecialties = potentialMentor.getSpecialties();
 
-        //see if new mentor is not already saved AND contains correct criteria
-        if(!(userSavedMatches.contains(potentialMentor)) && (mentorSpecialties.contains(matchCriteria))) {
-            mentorMatches.add(potentialMentor);
-        }
+      boolean isCurrentUser = potentialMentor.equals(currentUser);
+      boolean isAlreadySaved = userSavedMatches.contains(potentialMentor);
+      boolean sharedSpecialty = mentorSpecialties.contains(matchCriteria);
+
+      //see if new mentor is not same as logged in user AND is not already saved AND contains correct criteria
+      if(!isCurrentUser && !isAlreadySaved && sharedSpecialty) {
+        mentorMatches.add(potentialMentor);
+      }
     }
 
-    List<User> rankedMatches = MatchRanking.rankMatches(userSavedMatches, allUsers, mentorMatches);
-    
+    Map<String, Integer> bioMap;
+    if(userRepository instanceof PersistentUserRepository) {
```

Comment:
I took this out in my latest changes, I had originally used it to differentiate from testing, but I was able to update it so that it doesn't need this